connectors/sources/confluence.py

```diff
@@ -313,6 +314,27 @@ def get_default_configuration(cls):
             },
         }
 
+    @classmethod
+    def features(cls):
+        return Features(
+            {
+                "sync_rules": {
+                    "basic": {
+                        "enabled": True,
+                    },
+                    "advanced": {
+                        "enabled": True,
+                    },
+                },
+                "document_level_security": {
+                    "enabled": False,
+                },
+                "incremental_sync": {
+                    "enabled": False,
+                },
+            }
+        )
+
```

Comment:
I'm a little worried about having this structure bleed into our source code. What will maintenance look like if we decide to change how we structure this object in `.elastic-connectors`? How would we be able to handle needing to change this structure in any custom-made connectors that we don't have in our own source-control?

Part of me thinks that we should have just a list of keys of features that _are_ supported, and let the framework take that list and format into the correct structure. Or another approach is that we could add a `supports_x_feature()` function to the BaseDataSource for each new feature, and have the default impl be false?

WDYT?

Reply:
>Part of me thinks that we should have just a list of keys of features that are supported, and let the framework take that list and format into the correct structure. Or another approach is that we could add a `supports_x_feature()` function to the BaseDataSource for each new feature, and have the default impl be false?

This is a good idea. Instead of `supports_x_feature()`, I probably will use class variables: https://github.com/elastic/connectors-python/pull/986#discussion_r1213385310

Reply:
Changes were done here: https://github.com/elastic/connectors-python/pull/1114/commits/50b420bc9ef3548918d5fdb7f00c8ce456406b77, which reduced the number of files changed to 12.